l10n_ch_invoice_reports/models/mail_template.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+import base64
+
+from odoo import _, models
+
+
+class MailTemplate(models.Model):
+    _inherit = "mail.template"
+
+    def generate_email(self, res_ids, fields=None):
+        """ Method overridden in order remove QR/ISR payslips and use
+            Invoice report with payslip generated by this module
+        """
+
+        result = super().generate_email(res_ids, fields)
+
+        multi_mode = True
+        if isinstance(res_ids, int):
+            res_ids = [res_ids]
+            multi_mode = False
+
+        res_ids_to_templates = self.get_email_template(res_ids)
+        for res_id in res_ids:
+            related_record = self.env[self.model_id.model].browse(res_id)
+            template = res_ids_to_templates[res_id]
+            if related_record._name == "account.move":
```

Comment:
This should be at the beginning of the function like this:
```suggestion
            if self._name == "account.move":
```
So the previous code is not run for non 'account.move' recordset.